dist/s_all

```diff
@@ -22,8 +22,28 @@ while len(pids):
 ' $$
 }
 
+CLRL=$'\e[K'     # Clear to the end of line.
+CLRS=$'\e[0J'    # Clear to the bottom of screen.
+BOL=$'\r'        # Move cursor to the beginning of line.
+# stty unfortunately uses stdin (not stdout) to locate the tty so have to check both.
+[[ -t 1 && -t 0 ]] && { ISTTY=1; CLRIFTTY="$CLRS"; }
```

Comment:
Should this be: [[ -t 1 || -t 0 ]] && { ISTTY=1; CLRIFTTY="$CLRS"; }

Reply:
@tod-johnson-mongodb Hey this line checks if both STDIN and STDOUT are terminals, then it turns on interactive status reporting. I hope it does it right.

Originally I only checked stdin because that's what I care about.
Unfortunately, it turns out that `stty` uses the file descriptor 0 (stdin) to access the terminal and it's not configurable. If stdin happens to be redirected then stty doesn't know how to access the tty and the whole thing breaks.
That's why I had to add `-t 0` to the condition.

If you know a relatively easy and crossplatform way to make `stty` work with the file descriptor 1, do let me know.

Cheers,
Yury

Reply:
I was wrong and you are right. Sorry.